src/plugins/plan/planInterceptor.js

```diff
@@ -0,0 +1,38 @@
+/*****************************************************************************
+ * Open MCT, Copyright (c) 2014-2021, United States Government
+ * as represented by the Administrator of the National Aeronautics and Space
+ * Administration. All rights reserved.
+ *
+ * Open MCT is licensed under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * Open MCT includes source code licensed under additional open source
+ * licenses. See the Open Source Licenses file (LICENSES.md) included with
+ * this source code distribution or the Licensing information page available
+ * at runtime from the About dialog for additional information.
+ *****************************************************************************/
+
+export default function planInterceptor(openmct, configuration) {
+
+    openmct.objects.addGetInterceptor({
+        appliesTo: (identifier, domainObject) => {
+            return domainObject && domainObject.type === 'plan';
+        },
+        invoke: (identifier, object) => {
+
+            if (object && configuration !== undefined && configuration.getState) {
+                openmct.status.set(identifier, configuration.getState(object));
```

Comment:
I think we can avoid some boilerplate code by just moving this interceptor into the viper-openmct repository as a plugin, since it is VIPER specific. If we find in future that we need a generic way of providing state evaluation as configuration we could consider doing it then.

My general philosophy with this sort of thing is to keep the surface area of our public API as small as possible to minimize maintenance, and only increase it when we need to. How do we know when we need to add something to our public API? Generally speaking if we need to do the same thing in a number of different places in our code then it should probably be promoted to API. I try to avoid speculative API that we think may be useful in the future until we absolutely need it. eg. It's killing me to [not fix the status API right now](https://github.com/nasa/openmct/issues/3551#issuecomment-962202298) 😄

Reply:
Good idea. I'll move it over.

Reply:
Done